code/perl/index_filters.pl

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+
+use JSON::MaybeXS;
+use Search::Xapian ':all';
+use FindBin qw($Bin);
+use lib $Bin;
+use Support;
+use Data::Dumper;
+
+my ($data_path, $db_path) = @ARGV;
+die "Usage $0 DATAPATH DBPATH" unless $data_path && $db_path;
+
+index_csv($data_path, $db_path);
+
+sub index_csv {
+    my ($data_path, $db_path) = @_;
+    # Create or open the database we're going to be writing to.
+    my $db = Search::Xapian::WritableDatabase->new($db_path, DB_CREATE_OR_OPEN);
+    # Set up a TermGenerator that we'll use in indexing.
+    my $term_generator = Search::Xapian::TermGenerator->new;
+    $term_generator->set_stemmer(Search::Xapian::Stem->new('en'));
+    foreach my $rec (Support::parse_csv($data_path)) {
+        # print Dumper($rec);
+        my $doc = Search::Xapian::Document->new;
+        $term_generator->set_document($doc);
+        # Index each field with a suitable prefix.
+        $term_generator->index_text($rec->{TITLE}, 1, 'S');
+        $term_generator->index_text($rec->{DESCRIPTION}, 1, 'XD');
+
+        ### Start of new indexing code.
+        # Index the MATERIALS field, splitting on semicolons.
+        foreach my $material (split(/;/, $rec->{MATERIALS})) {
+            $material =~ s/\A\s*//;
+            $material =~ s/\z\s*//;
```

Comment:
This removes whitespace **after** the end of the string - I think you meant `s/\s*\z//`.